cpp/include/raft/util/cache_util.cuh

```diff
@@ -50,7 +50,7 @@ __global__ void get_vecs(
   if (tid < n_vec * n) {
     size_t out_col   = tid / n_vec;  // col idx
     size_t cache_col = cache_idx[out_col];
-    if (cache_idx[out_col] >= 0) {
+    if (!std::is_signed<idx_t>::value || cache_idx[out_col] >= 0) {
```

Comment:
```suggestion
    if constexpr (!std::is_signed<idx_t>::value || cache_idx[out_col] >= 0) {
```

Reply:
That does not compile because `cache_idx[out_col]` is not compile time constant.

Reply:
Ah. You can write it as two if-conditions then, write? `if constexpr` as the outer if and then the inner condition which is a plain `if`. Anyway, this isn't blocking by any means. I will leave it up to you to decide what way you want to do it.